packages/plugin-koa/src/koa.js

```diff
@@ -27,7 +31,21 @@ module.exports = {
         event.addMetadata('request', metadata)
       }, true)
 
+      // unhandled errors caused by this request
+      dom.on('error', (err) => {
+        const event = client.Event.create(err, false, handledState, 'koa middleware', 1)
+        ctx.bugsnag._notify(event, () => {}, (e, event) => {
+          if (e) client._logger.error('Failed to send event to Bugsnag')
+          ctx.bugsnag._config.onUncaughtException(err, event, client._logger)
+        })
+        if (!ctx.response.headersSent) {
+          ctx.response.status = 500
+        }
+      })
+
+      dom.enter()
       await next()
+      dom.exit()
```

Comment:
using domain in koa (as with express and restify) means that we are able to add request context to a wider class of error situation